### PR TITLE
Implement NARX

### PR DESCRIPTION
**Pull request by muguryalcin**
Implemented NARX and its test

### CODE REVIEW

File: chronokit/narx/narx.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    
+    def fit(self, epochs=20, lr=0.001):
+        # Loading data and model to device
+        self.dataloader = self._load_data()
+        self.model = self.model.to(self.device)
+
+        # Training
+        train_hist = []
+        criterion = torch.nn.MSELoss()
+        optimizer = torch.optim.AdamW(self.model.parameters(),lr=0.01)
```
Comment (hasanemirakin) [inline chronokit/narx/narx.py:64]: We might want to take the loss function and optimizer used as arguments also

Comment (hasanemirakin) [reply chronokit/narx/narx.py:64]: It would be good to write another function that will initialize the config to use in training such as loss function, optimizer, gradient clipping etc...

File: chronokit/narx/narx.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import torch
+import torch.nn as nn
+from chronokit.base._models import Model
+from chronokit.preprocessing._dataloader import DataLoader
+
+
+class NARXModel(torch.nn.Module):
+
+    def __init__(self, n_features, hidden_size):
```
Comment (hasanemirakin) [inline chronokit/narx/narx.py:9]: We might need to allow usage of regularization like dropout layers, batch normalization etc... if the user desires

File: chronokit/narx/narx.py (hunk 1 above)
Comment (muguryalcin) [reply chronokit/narx/narx.py:64]: Started working on those features.

**Comment by muguryalcin** [comment]
Added config support with batch norm and dropout.

**Review by hasanemirakin — APPROVED** [review]
Eline sağlık
